src/lib/Builder.hs

```diff
@@ -296,21 +295,21 @@ addInstanceSynthCandidate :: TopBuilder m => ClassName n -> InstanceName n -> m
 addInstanceSynthCandidate className instanceName =
   emitSynthCandidates $ SynthCandidates [] (M.singleton className [instanceName])
 
-emitAtomRules :: TopBuilder m => AtomName n -> AtomRules n -> m n ()
+emitAtomRules :: TopBuilder m => AtomName r n -> AtomRules n -> m n ()
 emitAtomRules v rules = emitNamelessEnv $
   TopEnvFrag emptyOutFrag $ mempty { fragCustomRules = CustomRules $ M.singleton v rules }
 
-emitTopLet :: (Mut n, TopBuilder m) => NameHint -> LetAnn -> Expr n -> m n (AtomName n)
+emitTopLet :: (Mut n, TopBuilder m) => NameHint -> LetAnn -> Expr r n -> m n (AtomName r n)
 emitTopLet hint letAnn expr = do
   ty <- getType expr
-  emitBinding hint $ AtomNameBinding $ LetBound  (DeclBinding letAnn ty expr)
+  emitBinding hint $ AtomNameBinding $ unsafeCoerceIRE $ LetBound (DeclBinding letAnn ty expr)
```

Comment:
Why is this coercion necessary?

Reply:
Because currently the top bindings are all `CoreIR`